plaidml/edsl/edsl.h

```diff
@@ -107,13 +113,43 @@ class Program {
   ///
   const std::vector<ProgramArgument>& outputs() const { return outputs_; }
 
+  ///
+  /// floatx
+  ///
+  const DType& floatx() const { return floatx_; }
+
+  void set_floatx(const DType& floatx) {
+    if (floatx != DType::FLOAT32 && floatx != DType::FLOAT64) {
+      std::stringstream ss;
+      ss << "Invalid floatx_ requested by Program. Expected either float32 or float64";
+      throw std::runtime_error(ss.str());
+    }
+    floatx_ = floatx;
+  }
+
+  ///
+  /// intx
+  ///
+  const DType& intx() const { return intx_; }
+
+  void set_intx(const DType& intx) {
+    if (intx != DType::INT32 && intx != DType::INT64) {
+      std::stringstream ss;
+      ss << "Invalid intx_ requested by Program. Expected either int32 or int64";
+      throw std::runtime_error(ss.str());
+    }
+    intx_ = intx;
+  }
+
```

Comment:
I was thinking that `floatx` and `intx` would be part of the built `Program` state and that we would avoid allowing mutating these parameters after the `Program` was constructed. What use case do you have in mind where you want to change them after the `Program` was constructed? Also, wouldn't we need to re-compile if these values were changed? Currently the compilation happens in the `Program` constructor.

Reply:
Removed these functions.